src/main/scala/ai/privado/passes/PropertyCollectorPass.scala

```diff
@@ -0,0 +1,391 @@
+package ai.privado.utility
+
+import io.shiftleft.codepropertygraph.generated.EdgeTypes
+import io.shiftleft.codepropertygraph.generated.nodes.{MethodParameterIn, NewJavaProperty}
+import overflowdb.BatchedUpdate
+import ai.privado.cache.RuleCache
+import io.joern.x2cpg.SourceFiles
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.{Literal, Member, NewFile, Call}
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import org.slf4j.LoggerFactory
+import io.shiftleft.semanticcpg.language._
+
+import java.io.File
+import scala.io.Source
+import java.util.Properties
+import scala.jdk.CollectionConverters._
+import io.circe.parser._
+import io.circe._
+
+import scala.collection.mutable
+
+import com.typesafe.config._
+
+import scala.xml.XML
+
+import com.github.wnameless.json.flattener.JsonFlattener
+import io.circe.yaml.parser
+import overflowdb.traversal._
+import ai.privado.model.Language
+
+class PropertyCollectorPass(cpg: Cpg, projectRoot: String, ruleCache: RuleCache, language: Language.Value)
+    extends ForkJoinParallelCpgPass[String](cpg) {
+
+  val logger = LoggerFactory.getLogger(getClass)
+
+  override def generateParts(): Array[String] = {
+    language match {
+      case Language.JAVA       => { configFiles(projectRoot, Set(".properties", ".yml", ".yaml", ".xml")).toArray }
```

Comment:
We can create Constant values for these extensions here at this file level, not in Constants.scala